src/rt-app.c

```diff
@@ -300,38 +304,49 @@ static int run_event(event_data_t *event, int dry_run,
 		pthread_mutex_unlock(&(rdata->res.barrier.m_obj));
 		break;
 	case rtapp_sig_and_wait:
-		log_debug("signal and wait %s", rdata->name);
+		log_debug("[%d] sync %s", task_idx, rdata->name);
+		log_debug("[%d] sync - signal %s", task_idx, rdata->name);
 		pthread_cond_signal(&(rdata->res.cond.obj));
+		log_debug("[%d] sync - wait %s", task_idx, rdata->name);
 		pthread_cond_wait(&(rdata->res.cond.obj), &(ddata->res.mtx.obj));
+		log_debug("[%d] sync -complete %s", task_idx, rdata->name);
 		break;
 	case rtapp_broadcast:
 		pthread_cond_broadcast(&(rdata->res.cond.obj));
 		break;
 	case rtapp_sleep:
 		{
 		struct timespec sleep = usec_to_timespec(event->duration);
-		log_debug("sleep %d ", event->duration);
-		nanosleep(&sleep, NULL);
+		log_debug("[%d] sleep %d (%ld.%09ld) start", task_idx, event->duration, sleep.tv_sec, sleep.tv_nsec);
+		while(1) {
```

Comment:
sleep.tv_sec and  sleep.tv_nsec are redundant with duration

Reply:
IIRC, the values of sleep are updated if a signal stops the sleep - this lets you know how much time passed before the code was interrupted.

Reply:
This is now a separate commit.